### PR TITLE
Split WebAPIKit into modules manually

### LINKED ISSUE #14: Multiple library products and union types
Playing with the library a bit more, I think it would be great to split it into separate targets and products, if possible. As an example, people who don't need WebGL don't need to build WebGL wrappers, but I think it still makes sense to provide them. 

IDK if WebIDL specifies dependencies between their "modules", so maybe we'd need to hardcode those on our side for things to work?

### PR DESCRIPTION
**Pull request by MaxDesiatov**
New `Module` type is added that specifies dependencies between modules manually.

More advanced approaches could be implemented in the future. For example, automated dependency resolution as investigated in https://github.com/swiftwasm/WebAPIKit/pull/30.

Partially resolves #14.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -1,20 +1,48 @@
-// swift-tools-version:5.3
+// swift-tools-version:5.5
```
Comment (MaxDesiatov) [inline Package.swift:1]: Bumped due to use `async` in the codebase.

File: Package.swift [hunk 2]
```diff
@@ -1,20 +1,48 @@
...
 // The swift-tools-version declares the minimum version of Swift required to build this package.
 
 import PackageDescription
 
 let package = Package(
     name: "WebAPIKit",
+    platforms: [.macOS(.v10_13)],
```
Comment (MaxDesiatov) [inline Package.swift:8]: Bumped due to use of certain file APIs in codegen sources.

**Comment by MaxDesiatov** [comment]
While this builds and runs without failure right now, it still needs tweaking to avoid duplication of property wrappers in strings across modules.

File: Sources/WebIDLToSwift/ModuleState.swift [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import WebIDL
+
+@dynamicMemberLookup
+enum ModuleState {
```
Comment (MaxDesiatov) [inline Sources/WebIDLToSwift/ModuleState.swift:4]: `Context` renamed to `ModuleState` here.

File: Sources/WebIDLToSwift/ModuleState.swift [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        closurePatterns.insert(closurePattern)
+    }
+
+    static func reset() {
+        unions = []
+        strings = ["toString"]
+    }
+}
+
+struct ScopedState {
```
Comment (MaxDesiatov) [inline Sources/WebIDLToSwift/ModuleState.swift:49]: Old `Context.State` becomes `ScopedState` here.

**Review by kateinoigakukun — APPROVED** [review]

**Pull request merged**